utils/build-script

```diff
@@ -1473,6 +1514,191 @@ details of the setups of other systems or automated environments.""")
+            # stdlib-target to build.
+            if host != args.host_target and host == stdlib_target:
+                return True
+
+        if host == args.host_target:
+            stdlib_targets = args.stdlib_deployment_targets
+        else:
+            stdlib_targets = [host]
+
+        swift_stdlib_targets = []
+        swift_sdks = set()
+        swift_benchmark_targets = []
+        swift_run_benchmark_targets = []
+        swift_test_targets = []
+        for stdlib_target_name in stdlib_targets:
+            # Get the target object.
+            stdlib_target = StdlibDeploymentTarget.get_target_for_name(
+                stdlib_target_name)
+            if stdlib_target is None:
+                diagnostics.fatal("unknown target: %r" % (stdlib_target_name,))
+
+            # Add the SDK to use.
+            swift_sdks.add(stdlib_target.platform.sdk_name)
+
+            # If we aren't building this target, do nothing else.
+            #
+            # FIXME: This matches build-script-impl behavior, but why add the
+            # SDK? Is that right?
+            if not should_build_stdlib_target(stdlib_target):
+                continue
+
+            # Compute build and test information.
+            build = stdlib_target.platform not in platforms_to_skip_build
+            test = stdlib_target.platform not in platforms_to_skip_test
+            test_host_only = None
+            build_benchmark = build and stdlib_target.supports_benchmark
+            # FIXME: Note, `build-script-impl` computed a property here
+            # w.r.t. testing, but it was actually unused.
+
+            # As a special case, Darwin embedded non-simulator platforms have
+            # different behavior for the test condition.
+            if stdlib_target.platform.is_darwin and \
+               stdlib_target.platform.is_embedded and \
+               not stdlib_target.platform.is_simulator:
+                if stdlib_target.platform not in platforms_to_skip_test_host:
+                    test_host_only = True
+                    test = True
+                else:
+                    test = False
+
+            name = stdlib_target.name
+            if build:
+                # FIXME: This logic doesn't make much sense, but
+                # build-script-impl had no comments.
+                if args.build_swift_stdlib_unittest_extra or \
+                       args.validation_test or args.long_test:
+                    swift_stdlib_targets.append("swift-stdlib-"+name)
+                else:
+                    swift_stdlib_targets.append("swift-test-stdlib-"+name)
+            if build_benchmark:
+                swift_benchmark_targets.append("swift-benchmark-"+name)
+                # FIXME: This probably should respect `args.benchmark`, but a
+                # typo (?) in build-script-impl meant we always would do this.
+                swift_run_benchmark_targets.append(
+                    "check-swift-benchmark-"+name)
```

Comment:
```sh
        if [[ "${build_benchmark_this_target}" ]] && [[ "${is_in_build_list}" ]]; then
            SWIFT_BENCHMARK_TARGETS+=("swift-benchmark-${stdlib_deployment_target}")
            if [[ $(not ${SKIP_TEST_BENCHMARK}) ]] ; then
              SWIFT_RUN_BENCHMARK_TARGETS+=("check-swift-benchmark-${stdlib_deployment_target}")
            fi
        fi
```

Looks like running the benchmarks was predicated on `--skip-test-benchmark` (but there is no such option, there is `--skip-test-benchmark*s*`).